sdk/storage/AzureStorageBlob/Source/StorageBlobClient.swift

```diff
@@ -39,30 +39,38 @@ public final class StorageBlobClient: PipelineClient {
     /// Options provided to configure this `StorageBlobClient`.
     public let options: StorageBlobClientOptions
 
-    /// The `TransferDelegate` to inform about transfer events.
+    /// The `TransferDelegate` to inform about events from transfers created by this `StorageBlobClient`.
     public weak var transferDelegate: TransferDelegate?
 
     private static let defaultScopes = [
         "https://storage.azure.com/.default"
     ]
 
-    fileprivate var managing = false
-    fileprivate lazy var manager: TransferManager = {
-        var manager = URLSessionTransferManager.shared
-        manager.delegate = self
-        manager.logger = self.logger
-        return manager
-    }()
+    private var managing = false
+    private let manager: TransferManager = URLSessionTransferManager.shared
+    private let restorationId: String
 
     // MARK: Initializers
 
     /// Create a Storage blob data client.
     /// - Parameters:
     ///   - baseUrl: Base URL for the storage account.
     ///   - authPolicy: An `Authenticating` policy to use for authenticating client requests.
+    ///   - restorationId: An identifier used to associate this client with transfers it creates. When a transfer is
+    ///     reloaded from disk (e.g. after an application crash), it can only be resumed once a client with the same
+    ///     `restorationId` has been initialized. If your application only uses a single `StorageBlobClient`, it is
+    ///     recommended to use a value unique to your application (e.g. "MyApplication"). If your application uses
+    ///     multiple clients with different configurations, use a value unique to both your application and the
+    ///     configuration (e.g. "MyApplication.userClient").
     ///   - options: Options used to configure the client.
-    private init(baseUrl: URL, authPolicy: Authenticating, withOptions options: StorageBlobClientOptions? = nil) {
+    private init(
+        baseUrl: URL,
+        authPolicy: Authenticating,
+        withRestorationId restorationId: String,
```

Comment:
I had a default value initially, but I decided to remove it. I like having a default for ergonomic reasons, but then I thought - what if the customer created a blob storage client using the default ID, but the customer happened to also use another third-party library and that library -also- created a storage client with the default ID. It would be easy for this to happen by accident, and it would cause major problems. So I went down the path of making the ID mandatory, but with API documentation that tries to explain what it's for and how to choose a good one.

Reply:
That makes sense. Other classes like OperationQueue and Logger require the same, so not too onerous.